htdocs/categories/class/categorie.class.php

```diff
@@ -1957,4 +1957,97 @@ public static function replaceThirdparty(DoliDB $db, $origin_id, $dest_id)
 
         return CommonObject::commonReplaceThirdparty($db, $origin_id, $dest_id, $tables, 1);
     }
+
+	/**
+	 * Return a list with all selected categories of a category filter box
+	 *
+	 * @param string	$type			The category type (e.g Categorie::TYPE_WAREHOUSE)
+	 * @return Array					A list with all selected categories (Note: "-2" is typical "without category")
+	 */
+	public static function getPost($type)
+	{
+		return GETPOST("search_category_".$type."_list", "array");
+	}
+
+	/**
+	 * Return the addtional SQL JOIN query for filtering a list by a category
+	 *
+	 * @param string	$type			The category type (e.g Categorie::TYPE_WAREHOUSE)
+	 * @param string	$rowIdName		The name of the row id inside the whole sql query (e.g. "e.rowid")
+	 * @return string					A additional SQL JOIN query
+	 */
+	public static function getFilterJoinQuery($type, $rowIdName)
```

Comment:
Does not worth to have a function just fo that. Above all, the content of this function depends too much of how is fforged the sql request by the caller. This part of code woul be better in list.php